src/program/services/streaming/cache.py

```
@@ -80,48 +81,120 @@ def __init__(self, cfg: CacheConfig) -> None:
+        async with self._lock:
+            with self._thread_lock:
+                yield
+
+    async def _initialize(self) -> None:
         # Lazy-rebuild index for any pre-existing files so size limits apply after restart
         try:
-            if (self.cfg.eviction or "LRU").upper() == "LRU":
-                self._initial_scan()
+            if self.cfg.eviction == "LRU":
+                await self._initial_scan()
         except Exception as e:
```

Ruff (0.14.3)

Do not catch blind exception: `Exception`

(BLE001)